src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,224 @@
+    requestAnimationFrame(() => {
+      updateCellOffsets();
+    });
+  }, [updateCellOffsets]);
+
+  return { cellOffsets, updateCellOffsets };
+};
+
+interface GetStickyClassNamesProps {
+  styles: { [key: string]: string };
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  isStuckToTheLeft: boolean;
+  isStuckToTheRight: boolean;
+}
+
+export function getStickyClassNames({
+  styles,
+  isSticky,
+  isLastStickyLeft,
+  isLastStickyRight,
+  isStuckToTheLeft,
+  isStuckToTheRight,
+}: GetStickyClassNamesProps) {
+  return clsx({
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tableRefObject,
+  visibleColumnsLength,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns || (stickyColumns.first === 0 && stickyColumns.last === 0);
+  const columnsLengthWithSelection = visibleColumnsLength + (hasSelection ? 1 : 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  // Compute left table padding
+  const table = tableRefObject.current;
+  const tableLeftPadding = table ? parseInt(getComputedStyle(table).paddingLeft) : 0;
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tableLeftPadding < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [
+    containerWidth,
+    noStickyColumns,
+    stickyColumns,
+    totalStickySpace,
+    visibleColumnsLength,
```

Comment:
Why `visibleColumnsLength` is needed?

Reply:
In case the user changes which columns are visible in the collection preferences. Here we ensure this gets re-calculated.

Reply:
Yes - I had commented and noticed that as I looked deeper into it. Also, the visible columns length can be the same but the widths change.